catalog/spatial/geocoding/spatial-geocoding-feature/src/main/java/org/codice/ddf/spatial/geocoding/feature/CatalogFeatureIndexer.java

```diff
@@ -114,33 +122,42 @@ private void createOrUpdateMetacardForFeature(SimpleFeature feature, boolean cre
   }
 
   private Metacard createMetacardForFeature(SimpleFeature feature) throws FeatureIndexingException {
-    Metacard metacard = new MetacardImpl();
+    Metacard metacard = new MetacardImpl(metacardType);
     String countryCode = feature.getID();
-    metacard.setAttribute(new AttributeImpl(Core.TITLE, countryCode));
 
-    String wkt = WKT_WRITER_THREAD_LOCAL.get().write((Geometry) feature.getDefaultGeometry());
-    metacard.setAttribute(new AttributeImpl(Core.LOCATION, wkt));
+    Object nameObject = feature.getAttribute(NAME_KEY);
+    if (nameObject instanceof String) {
+      metacard.setAttribute(new AttributeImpl(Core.TITLE, (String) nameObject));
+
+      metacard.setAttribute(new AttributeImpl(Location.COUNTRY_CODE, countryCode));
+      String wkt = WKT_WRITER_THREAD_LOCAL.get().write((Geometry) feature.getDefaultGeometry());
+      metacard.setAttribute(new AttributeImpl(Core.LOCATION, wkt));
+
+      List<Serializable> tags =
+          Arrays.asList(GeoCodingConstants.DEFAULT_TAG, GeoCodingConstants.COUNTRY_TAG);
+      metacard.setAttribute(new AttributeImpl(Core.METACARD_TAGS, tags));
+      return metacard;
+    }
 
-    List<Serializable> tags =
-        Arrays.asList(GeoCodingConstants.DEFAULT_TAG, GeoCodingConstants.COUNTRY_TAG);
-    metacard.setAttribute(new AttributeImpl(Core.METACARD_TAGS, tags));
-    return metacard;
+    throw new FeatureIndexingException("Malformed feature");
   }
 
   private Metacard findMetacardForFeature(SimpleFeature feature) throws FeatureIndexingException {
-    String countryCode = feature.getID();
-
-    QueryRequest queryRequest =
-        new QueryRequestImpl(catalogHelper.getQueryForCountryCode(countryCode));
-    try {
-      SourceResponse response = catalogFramework.query(queryRequest);
-      if (response.getResults().isEmpty()) {
-        return null;
+    Object nameObject = feature.getAttribute(NAME_KEY);
+    if (nameObject instanceof String) {
+      String name = (String) feature.getAttribute(NAME_KEY);
```

Comment:
nit: cast `nameObject` to string instead of getting the attribute again

Reply:
fixed